utils/download_filtered.js

```diff
@@ -26,95 +36,121 @@ function downloadFiltered(config, callback) {
       minTime: 6000
     };
     const limiter = new Bottleneck(options);
-    const callbackOnLastOne = () => {
+    const done = () => {
       if (limiter.empty()) {
         callback();
       }
     };
-    files.map(file => {
-      limiter.submit(downloadSource, targetDir, file, callbackOnLastOne);
+    sources.map(source => {
+      limiter.submit(downloadSource, targetDir, source, done);
     });
     process.on('SIGINT', () => {
-      limiter.stop({dropWaitingJobs: true});
+      limiter.stop({ dropWaitingJobs: true });
       process.exit();
     });
   });
 
 }
 
-function getFiles(config, targetDir, main_callback){
-  const errorsFatal = config.get('imports.openaddresses.missingFilesAreFatal');
-  const files = config.imports.openaddresses.files;
-  files.forEach(file => {
-    // sources MUST end with '.csv'
-    if( !file.endsWith('.csv') ){
-      const msg = `invalid source '${file}': MUST end with '.csv'`;
-      logger.warn(msg);
-
-      // respect 'imports.openaddresses.missingFilesAreFatal' setting
-      return main_callback(errorsFatal ? msg : null);
+async function getSources(files) {
+  const OpenAddressesAPI = require('./OpenAddressesAPI');
+  const oa = new OpenAddressesAPI();
+
+  return await Promise.all(files.map(async file => {
+
+    // source definitions previously required a file extension.
+    // please remove file extensions from your ~/pelias.json file
+    // to silence these warning messages.
+    let id = file.replace(/\.[^\/.]+$/, '');
+    if (id !== file) {
+      logger.warn(`source definitions no longer require a file extension '${file}'`);
     }
-  });
-  return files.map(file => {
-    const source = file.replace('.csv', '.zip');
-    const name = file.replace('.csv', '').replace(/\//g,'-');
-    return {
-      csv: file,
-      url: `https://results.openaddresses.io/latest/run/${source}`,
-      zip: temp.path({prefix: name, dir: targetDir, suffix: '.zip'})
-    };
-  });
+
+    // lookup the source using the OpenAddresses API
+    // to find the most current job id and ensure validity
+    const version = await oa.lookup(id);
+    const valid = (version && version.job);
+
+    // invalid source
+    if (!valid) {
+      return { id, error: `invalid source '${file}'` };
+    }
+
+    // valid source
+    return { id, url: oa.url(version.job) };
+  }));
 }
 
-function downloadSource(targetDir, file, main_callback) {
+function downloadSource(targetDir, source, done) {
+
   const errorsFatal = config.get('imports.openaddresses.missingFilesAreFatal');
+  const token = config.get('imports.openaddresses.token');
   const referer = config.get('imports.openaddresses.dataReferer') || 'https://pelias-results.openaddresses.io';
-  logger.info(`Downloading ${file.csv}`);
+  logger.info(`Downloading ${source.id}`);
+
+  const outFile = path.join(targetDir, `${source.id}.geojson`);
+  const tmpFile = temp.path({
+    prefix: source.id.replaceAll(path.sep, '-'),
```

Comment:
`replaceAll` is available since node 15, maybe too soon to use it ?
Current matrix of pelias: 12.x; 14.x; 16.x

Reply:
resolved via rebase